### ISSUE TITLE
Search box when it redirect to search result page give error 

### ISSUE THREAD

**Issue by spo365** [microsoft-search/pnp-modern-search#195]
**Version used**
Ex: 3.7

**Describe the bug**
Search box when it redirect to search result page give error ()

**To Reproduce**
1. Add search box webpart and redirect new page to results page
2. give below error (Screenshots and error msg included)
**Expected behavior**
it should load the results 

**Screenshots**
If the problem persists, contact the site administrator and give them the information in Technical Details.
Technical Details

ERROR:
Object doesn't support property or method 'find'

CALL STACK:
TypeError: Object doesn't support property or method 'find'
   at Anonymous function (https://TEST.sharepoint.com/sites/testApps/ClientSideAssets/890affef-33e0-4d72-bd72-36399e02143b/search-box-web-

part_02d7824830cf38727b69de85ac8900ac.js:1:674386)
   at Anonymous function (https://TEST.sharepoint.com/sites/testApps/ClientSideAssets/890affef-33e0-4d72-bd72-36399e02143b/search-box-web-

part_02d7824830cf38727b69de85ac8900ac.js:1:674143)

**Desktop (please complete the following information):**
 - Browser  chrome, IE

**Additional context**
![image](https://user-images.githubusercontent.com/5062256/79627750-50390d00-817e-11ea-99a6-1261a32e59f7.png)

**Comment by wobba** [comment]
Can you try with v3.10 or newer? You should always test with a newer version for bugs found in older versions.

**Comment by spo365** [comment]
I have tried the latest version still getting the same error even when add the search box webpart. it show same error.


Something went wrong
If the problem persists, contact the site administrator and give them the information in Technical Details.
Technical Details

ERROR:
Object doesn't support property or method 'find'

CALL STACK:
TypeError: Object doesn't support property or method 'find'
   at Anonymous function (https://caltexau.sharepoint.com/sites/CaltexApps/ClientSideAssets/890affef-33e0-4d72-bd72-36399e02143b/search-box-web-part_02d7824830cf38727b69de85ac8900ac.js:1:674386)
   at Anonymous function (https://caltexau.sharepoint.com/sites/CaltexApps/ClientSideAssets/890affef-33e0-4d72-bd72-36399e02143b/search-box-web-part_02d7824830cf38727b69de85ac8900ac.js:1:674143)
   at Anonymous function (https://caltexau.sharepoint.com/sites/CaltexApps/ClientSideAssets/890affef-33e0-4d72-bd72-36399e02143b/search-box-web-part_02d7824830cf38727b69de85ac8900ac.js:1:667883)
   at Anonymous function (https://caltexau.sharepoint.com/sites/CaltexApps/ClientSideAssets/890affef-33e0-4d72-bd72-36399e02143b/search-box-web-part_02d7824830cf38727b69de85ac8900ac.js:1:667196)
   at s (https://caltexau.sharepoint.com/sites/CaltexApps/ClientSideAssets/890affef-33e0-4d72-bd72-36399e02143b/search-box-web-part_02d7824830cf38727b69de85ac8900ac.js:1:666757)
   at s (https://caltexau.sharepoint.com/sites/CaltexApps/ClientSideAssets/890affef-33e0-4d72-bd72-36399e02143b/search-box-web-part_02d7824830cf38727b69de85ac8900ac.js:1:21965)
   at Anonymous function (https://caltexau.sharepoint.com/sites/CaltexApps/ClientSideAssets/890affef-33e0-4d72-bd72-36399e02143b/search-box-web-part_02d7824830cf38727b69de85ac8900ac.js:1:22112)
   at c (https://caltexau.sharepoint.com/sites/CaltexApps/ClientSideAssets/890affef-33e0-4d72-bd72-36399e02143b/search-box-web-part_02d7824830cf38727b69de85ac8900ac.js:1:26109)

**Comment by spo365** [comment]
I have tried the latest version still getting the same error even when add the search box webpart. it show same error.

Something went wrong
If the problem persists, contact the site administrator and give them the information in Technical Details.
Technical Details
ERROR:
Object doesn't support property or method 'find'

**Comment by wobba** [comment]
@spo365 are you able to run in debug mode and get the exact line. If you had said IE only I would have thought polyfill issue, but since you also report chrome it seems a null ref somehow.

**Comment by spo365** [comment]
I am getting error on IE Object doesn't support property or method 'find' however on chrome it is showing with query parameter (when it redirect to result page ) "Error details: Error: ***Error initializing application. Error: TypeError: Cannot read property 'Search' of undefined"

**Comment by vamsi981** [comment]
I am also getting the same error 
Error: ***Error initializing application. Error: TypeError: Cannot read property 'Search' of undefined

**Comment by vamsi981** [comment]
i checked through debugger tools and i got the below error message
messageSource: "AUGURY_INSPECTED_APPLICATION"
messageId: "cab7c0517fdcb"
serialize: 0
messageType: 9

**Comment by wobba** [comment]
@vamsi981 that's not very helpful ;) If you run locally with `gulp serve`, you should get error in the .ts file where the error appears. But we'll try to reproduce when time and look at this.

**Comment by spo365** [comment]
In order to resolve IE issue we have replaced find with filter in searchwebpart.ts code line:-360
existingSavedProvider:any = savedProviders.filter(sp => sp.providerName === provider.providerName);

but query parameter is not working when it redirect to new page.
Error details: Error: ***Error initializing application. Error: TypeError: Cannot read property 'Search' of undefined

**Comment by wobba** [comment]
I'm fixing all the `.find` places in the code.

**Comment by wobba** [comment]
Duplicate of #124

**Comment by spo365** [comment]
Hi @wobba,

we are still get error while redirecting search to search result page. Is something because i am not using Office 365 CDN ?

Note this happening in all the browser and it is intermittent issue.
Error message :- 
![image](https://user-images.githubusercontent.com/5062256/81889474-f265dc80-95e6-11ea-9a8d-b2400e79c56b.png)

**Comment by MakingThingsWorkToronto** [comment]
Is your tenant on standard release?

I bet Microsoft is rolling out the new site search functionality To standard release... this error is in Microsoft code & I isolated on another client to the query string parameter q.

Can you please test a page that does not use the  PnP search webparts by appending ?q=* to the URL Please let me know if you see the same error.

The work around is to navigate to the page with the search results using a different query string - I used qq=* (it’ll load fine). Modify the web parts to use that query string parameter.. You’ll also need to replace the query string parameter on all callees.

Please let me know if this helps!

**Comment by spo365** [comment]
Yes my tenant is on standard release, now this issue is resolved automatically.
It's very confusing may i will try using qq in query parameter also i have request provide me log details.

**Comment by wobba** [comment]
We need a proper error message to trouble shoot @spo365. And is it only on IE11, cause the latest screenshot seems different from your original error. Does it only happen on redirect? Reproduceable on other tenants?

**Comment by spo365** [comment]
it is happening in my tenant only after adding @MakingThingsWorkToronto recommendation it is working fine. In my tenant CDN is not available i believe this should not be related to this ?

**Comment by MakingThingsWorkToronto** [comment]
I had the same error & once the search box appears in the top bar you can change the query string back to q. The error is caused because the new search box ui component isn’t on the page but the new core search script is. The new core search script is missing a null check. If the error was in the Modern Search SpFx webpart we would see a graceful failure not this obtrusive redirect to an error page... Noce to have the info listed here but not a bug in this code :) I’d close this @wobba ... Thanks for reporting!!.
